### PR TITLE
Change default solver to scipy

### PR DESCRIPTION
**Pull request by tsmbland**
# Description

Revisits the changes is #378 to change the default solver to scipy.

I think now is a good time to do this since v1.3 already has lots of changes that will require people to properly read the release notes and make changes to their settings files. If anyone misses this, that's their fault.

I've also added a warning if `lpsolver` is not specified in their settings file, to tell people that it's defaulting to scipy.

I had to change one of the tests as it was failing with the scipy solver. Not sure why (or why this wasn't a problem in #378), but switching to adhoc for this one test allows it to run.

### CODE REVIEW

File: docs/inputs/toml.rst [hunk 1]
```diff
@@ -442,12 +442,12 @@ Sectors contain a number of subsections:
         are registered via :py:func:`~muse.investments.register_investment`. At time of
         writing, three are available:
 
+        - "scipy" solver (default): Formulates investment as a true LP problem and solves it using
```
Comment (dalonsoa) [inline docs/inputs/toml.rst:445]: Maybe indicate from when it is the default solver?
```suggestion
        - "scipy" solver (default from v1.3): Formulates investment as a true LP problem and solves it using
```

File: docs/inputs/toml.rst [hunk 2]
```diff
@@ -442,12 +442,12 @@ Sectors contain a number of subsections:
...
+          the `scipy solver`_.
+
         - an "adhoc" solver: Simple in-house solver that ranks the technologies
           according to cost and service the demand incrementally.
 
-        - "scipy" solver: Formulates investment as a true LP problem and solves it using
-          the `scipy solver`_.
-
         - "cvxopt" solver: Formulates investment as a true LP problem and solves it
```
Comment (dalonsoa) [inline docs/inputs/toml.rst:451]: Is this actually working? I've never tried it or had to deal with it.

**Review by dalonsoa — APPROVED** [review]
LGTM!

File: docs/inputs/toml.rst (hunk 2 above)
Comment (tsmbland) [reply docs/inputs/toml.rst:451]: A quick test suggests the answer is no (at least in the current MUSE version - it's not covered by the tests so might have been accidentally broken along the way)

**Pull request merged**